File: src/test/java/com/raychenon/leetcode/GenerateParenthesesTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2019-11-05
 * https://leetcode.com/problems/generate-parentheses/
 */
public class GenerateParenthesesTest {

    private static GenerateParenthesis instance;

    @BeforeClass
    public static void setUp() {
        instance = new GenerateParenthesis();
    }

    @Test
    public void size1Test() {
        List<String> expectedResults = Arrays.asList("()");
        assertEquals(expectedResults, instance.generateParenthesis(1));
    }

    @Test
    public void size3Test() {
        List<String> expectedResults = Arrays.asList("((()))",
                "(()())",
                "(())()",
                "()(())",
                "()()()");
        assertEquals(expectedResults, instance.generateParenthesis(3));
    }


}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/GenerateParenthesesTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.Assert.assertEquals
import junit.framework.Assert.assertTrue
import org.junit.Test

/**
 * User: raychenon
 * Date: 2019-11-05
 * https://leetcode.com/problems/generate-parentheses/
 */
class GenerateParenthesesTest {

    @Test
    fun size1Test() {
        val expectedResults = listOf("()")
        assertEquals(expectedResults, GenerateParentheses.generateParenthesisBacktracking(1))
        assertEquals(expectedResults, GenerateParentheses.generateParenthesisClosureNumber(1))
    }

    @Test
    fun size3Test() {
        val expectedResults = listOf(
            "((()))",
            "(()())",
            "(())()",
            "()(())",
            "()()()"
        )
        assertEquals(expectedResults, GenerateParentheses.generateParenthesisBacktracking(3))
        assertTrue(GenerateParentheses.generateParenthesisClosureNumber(3).containsAll(expectedResults))
    }
}